test/test_compress.cpp

```diff
@@ -35,7 +35,7 @@ TEST_CASE("Compress index", "[index][compress]") {
         ScorerParams(""),  // no scorer
         std::nullopt,  // no quantization
         true,  // check=true
-        false  // in-memory=false
+        GENERATE(true, false)  // in-memory=(true, false)
```

Comment:
Can we move it to a separate variable, just like encoding? Then we can have `in_memory` here and be can remove the comment.

Note that these comments are only necessary because of a somewhat badly designed API. Hopefully we can refactor it so that it's always clear what argument we're passing. Just a side note.